libs/langchain/langchain/vectorstores/opensearch_vector_search.py

```diff
@@ -494,15 +527,28 @@ def _raw_similarity_search_with_score(
         search_type = _get_kwargs_value(kwargs, "search_type", "approximate_search")
         vector_field = _get_kwargs_value(kwargs, "vector_field", "vector_field")
 
+        if (
+            self.is_aoss
+            and search_type != "approximate_search"
+            and search_type != SCRIPT_SCORING_SEARCH
+        ):
+            raise ValueError(
+                "AOSS only supports `approximate_search` and `script_scoring`"
+            )
+
         if search_type == "approximate_search":
             boolean_filter = _get_kwargs_value(kwargs, "boolean_filter", {})
             subquery_clause = _get_kwargs_value(kwargs, "subquery_clause", "must")
+            efficient_filter = _get_kwargs_value(kwargs, "efficient_filter", {})
+            # `lucene_filter` is deprecated, added for Backwards Compatibility
             lucene_filter = _get_kwargs_value(kwargs, "lucene_filter", {})
-            if boolean_filter != {} and lucene_filter != {}:
+
+            if boolean_filter != {} and efficient_filter != {}:
```

Comment:
should this check be there for lucene filters also? I know it's deprecated, but we need to put that check in deprecation case too

Reply:
ack